user/super/com/google/gwt/emul/java/math/BigInteger.java

```diff
@@ -260,16 +260,42 @@ private static void setFromString(BigInteger bi, String val, int radix) {
    * @throws NumberFormatException if the length of {@code val} is zero.
    */
   public BigInteger(byte[] val) {
+    this(val, 0, val.length);
+  }
+
+  /**
+   * Constructs a new {@code BigInteger} from the given two's complement
+   * representation. The most significant byte is the entry at index 0. The most
+   * significant bit of this entry determines the sign of the new {@code
+   * BigInteger} instance. The given array must not be empty.
+   *
+   * @param val two's complement representation of the new {@code BigInteger}.
+   * @param offset the start offset of the binary representation.
+   * @param length the number of bytes to use.
+   * @throws NullPointerException if {@code val == null}.
+   * @throws NumberFormatException if the length of {@code val} is zero.
+   */
+  public BigInteger(byte[] val, int offset, int length) {
     if (val.length == 0) {
       // math.12=Zero length BigInteger
       throw new NumberFormatException("Zero length BigInteger"); //$NON-NLS-1$
     }
-    if (val[0] < 0) {
+    if (length < 0 || offset < 0 || val.length < 0 || length > val.length - offset) {
```

Comment:
```suggestion
    if (length < 0 || offset < 0 || length > val.length - offset) {
```
We shouldn't ever need the array length check